framework/src/problems/FEProblemBase.C

```diff
@@ -573,6 +574,10 @@ FEProblemBase::initialSetup()
 {
   TIME_SECTION(_initial_setup_timer);
 
+  if (_skip_exception_check)
+    mooseWarning(
+        "Useful information will be ignored if code crash because you skip exception check");
```

Comment:
Your warning isn't very descriptive of what's happening. Can you perhaps replace the message with this?

"MOOSE may fail to catch an exception when the "skip_exception_check" parameter is used. If you receive a terse MPI error during execution, remove this parameter and rerun your simulation."